config/configtls/configtls_test.go

```diff
@@ -201,3 +206,166 @@ func TestLoadTLSServerConfig(t *testing.T) {
+		assert.Equal(t, "mkcert tristan@tristans", pCert.Issuer.CommonName)
+
+		// Change the certificate
+		assert.NoError(t, certFile.Truncate(0))
+		assert.NoError(t, keyFile.Truncate(0))
+		_, err = certFile.Seek(0, 0)
+		assert.NoError(t, err)
+		_, err = keyFile.Seek(0, 0)
+		assert.NoError(t, err)
+
+		fdc2, err := os.Open(path.Join("testdata", "test-cert2.pem"))
+		assert.NoError(t, err)
+		_, err = io.Copy(certFile, fdc2)
+		assert.NoError(t, err)
+		assert.NoError(t, fdc2.Close())
+
+		fdk2, err := os.Open(path.Join("testdata", "test-key2.pem"))
+		assert.NoError(t, err)
+		_, err = io.Copy(keyFile, fdk2)
+		assert.NoError(t, err)
+		assert.NoError(t, fdk2.Close())
+
+		// Wait ReloadInterval to ensure a reload will happen
+		time.Sleep(options.ReloadInterval)
+
+		// Assert that we loaded the new certificate
+		cert, err = cfg.GetCertificate(&tls.ClientHelloInfo{})
+		assert.NoError(t, err)
+		assert.NotNil(t, cert)
+		pCert, err = x509.ParseCertificate(cert.Certificate[0])
+		assert.NoError(t, err)
+		assert.NotNil(t, pCert)
+		assert.Equal(t, "example.net", pCert.Issuer.CommonName)
+	})
+
+	t.Run("Should not reload the certificate if reload-interval is 0", func(t *testing.T) {
+		// Copy certs into a temp dir so we can safely modify them
+		certFile, err := os.CreateTemp("", "cert")
+		assert.NoError(t, err)
+		defer os.Remove(certFile.Name())
+
+		keyFile, err := os.CreateTemp("", "key")
+		assert.NoError(t, err)
+		defer os.Remove(keyFile.Name())
+
+		fdc, err := os.Open(path.Join("testdata", "test-cert.pem"))
+		assert.NoError(t, err)
+		_, err = io.Copy(certFile, fdc)
+		assert.NoError(t, err)
+
+		fdk, err := os.Open(path.Join("testdata", "test-key.pem"))
+		assert.NoError(t, err)
+		_, err = io.Copy(keyFile, fdk)
+		assert.NoError(t, err)
+		assert.NoError(t, fdk.Close())
+
+		options := TLSSetting{
+			CertFile:       certFile.Name(),
+			KeyFile:        keyFile.Name(),
+			ReloadInterval: 0,
+		}
+		cfg, err := options.loadTLSConfig()
+		assert.NoError(t, err)
+		assert.NotNil(t, cfg)
+
+		// Asssert that we loaded the original certificate
+		cert, err := cfg.GetCertificate(&tls.ClientHelloInfo{})
+		assert.NoError(t, err)
+		assert.NotNil(t, cert)
+		pCert, err := x509.ParseCertificate(cert.Certificate[0])
+		assert.NoError(t, err)
+		assert.NotNil(t, pCert)
+		assert.Equal(t, "mkcert tristan@tristans", pCert.Issuer.CommonName)
+
+		// Change the certificate
+		assert.NoError(t, certFile.Truncate(0))
+		assert.NoError(t, keyFile.Truncate(0))
+		_, err = certFile.Seek(0, 0)
+		assert.NoError(t, err)
+		_, err = keyFile.Seek(0, 0)
+		assert.NoError(t, err)
+
+		fdc2, err := os.Open(path.Join("testdata", "test-cert2.pem"))
+		assert.NoError(t, err)
+		_, err = io.Copy(certFile, fdc2)
+		assert.NoError(t, err)
+		assert.NoError(t, fdc2.Close())
+
+		fdk2, err := os.Open(path.Join("testdata", "test-key2.pem"))
+		assert.NoError(t, err)
+		_, err = io.Copy(keyFile, fdk2)
+		assert.NoError(t, err)
+		assert.NoError(t, fdk2.Close())
```

Comment:
A good chunk of this and the previous tests are the same, perhaps split it into its own func? Like, "refreshCerts" ?

Reply:
Sure, I can refactor this to look more like a table test.